proposals/0000-local-import.rst

```diff
@@ -0,0 +1,225 @@
+Local import
+==============
+
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+            **After creating the pull request, edit this file again, update the
+            number in the link, and delete this bold sentence.**
+.. sectnum::
+.. contents::
+
+This proposal allows the use of ``import`` statements in any ``let``, ``where`` or ``do`` block. The effect of those imports is limited to the corresponding block and (for ``let`` and ``where`` blocks) to the defined expression. Moreover, it defines a shortcut syntax, ``Qualifier.{ … }``, to allow scoping a local import over a single expression.
+
+For example, the following snippets are considered valid:
+
+::
+
+  module MyModule where
+
+  import Data.Set (Set)
+  import qualified Data.Set as Set
+  
+  foo :: (Set Int, Set String)
+  foo = (s1, fromList ["foo", "bar", "baz"])
+    where
+      import Set
+      s1 = fromList [1, 2, 3]
+
+::
+
+  module Main where
+  import System.Environment as Env
+   
+  main :: IO ()
+  main = do
+    args <- Env.getArgs
+    case args of
+      x:y:_ -> …
+      _ -> do
+        import qualified System.Exit as Exit
+        Exit.{ exitWith $ ExitFailure (-1) }
+
+Motivation
+------------
+This proposal is strongly inspired by two OCaml constructs, introduced in 2011 with version 3.12: ``let open Module in …`` and ``Module.( … )``. In less than 10 years, this feature has gained widespread adoption throughout the ecosystem, bringing significant benefits in terms of readability and maintainability.
```

Comment:
Couple of nitpicks here:

- This is not the right place for this. You can put it at the end of the summary (that's what I'd do) or at the end of the motivation section. But at any rate, it is a reference, not a starting point.
- I'd say something like « strongly inspired by Ocaml's “local open” feature which introduces two constructs[…] ».